src/browser/atom-portable.coffee

```diff
@@ -0,0 +1,32 @@
+fs = require 'fs-plus'
+path = require 'path'
+ipc = require 'ipc'
+
+module.exports =
+class AtomPortable
+  @getPortableAtomHomePath: ->
+    execDirectoryPath = path.dirname(process.execPath)
+    path.join(execDirectoryPath, '..', '.atom')
+
+  @isPortableInstall: (platform, environmentAtomHome, defaultHome) ->
+    return false unless platform is 'win32'
+    return false if environmentAtomHome
+    return false if not fs.existsSync(@getPortableAtomHomePath())
+    # currently checking only that the directory exists  and is writable,
+    # probably want to do some integrity checks on contents in future
+    @portableAtomHomePathWritable(defaultHome)
+
+  @portableAtomHomePathWritable: (defaultHome) ->
```

Comment:
Could you rename this to `isPortableAtomHomePathWritable` to match the style of `isPortableInstall`?